src/lib/sputter/net/port.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+namespace sputter { namespace net { class UDPPort {
+    public:
+        UDPPort();
+        ~UDPPort();
+    
+        bool open(uint16_t port);
+        // Close is automatically called in the destructor but can be called manually here.
+        void close();
+    
+        bool send(const void *data, int dataSize, const std::string &address, uint16_t port);
+        int receive(void *data, int dataSize, std::string &address, uint16_t &port);
```

Comment:
Another stylistic preference on my end, but I prefer that out parameters be pointers rather than references-- it's a lot more obvious at the call site that the value you're providing can be changed. e.g.

```C
// Passing the address of address and port makes the intent a bit clearer !
const int received = port.receive(data, dataSize, &address, &port); 
```

Reply:
Just occurred to me also-- I haven't made much use of them (completely unused in this codebase actually) but C++17 does support multiple return values